crates/re_space_view/src/data_query_blueprint.rs

```diff
@@ -384,25 +244,25 @@ impl<'a> QueryExpressionEvaluator<'a> {
         overrides: &EntityOverrides,
         inherited: &EntityProperties,
         data_results: &mut SlotMap<DataResultHandle, DataResultNode>,
-        from_recursive: bool,
     ) -> Option<DataResultHandle> {
         // If we hit a prefix that is not allowed, we terminate. This is
-        // a pruned branch of the tree. Can come from either an explicit
-        // recursive exclusion, or an implicit missing inclusion.
+        // a pruned branch of the tree.
+
+        // TODO(emilk): implement this optimization
+        // if self
+        //     .entity_path_filter
+        //     .is_everything_starting_with_excluded(&tree.path)
+        // {
+        //     return None;
+        // }
+
         // TODO(jleibs): If this space is disconnected, we should terminate here
-        if self.recursive_exclusions.contains(&tree.path)
-            || !(from_recursive || self.allowed_prefixes.contains(&tree.path))
-        {
-            return None;
-        }
 
         let entity_path = tree.path.clone();
 
         // Pre-compute our matches
-        let exact_include = self.exact_inclusions.contains(&entity_path);
-        let recursive_include = self.recursive_inclusions.contains(&entity_path) || from_recursive;
-        let exact_exclude = self.exact_exclusions.contains(&entity_path);
-        let any_match = (exact_include || recursive_include) && !exact_exclude;
+        let exact_include = self.entity_path_filter.is_exact_included(&entity_path);
```

Comment:
@jleibs I don't quite follow the logic below in this function. Why was the old code checking `exact_inclusions`? Do we care if the hit on `/car` is because of a `+ /car` rule or a `+ /car/**` rule?

Reply:
Maybe? My thought is that if you **explictly** added an entity based on a path, that entity should show up in the result-set (for things like the tree-UI), even if it has no data associated with it (we show id subdued in the menu in that case). That seems like a better user-experience for a user working with an existing blueprint and an RRD that might be missing data.